src/xcode/ENA/ENA/Source/Scenes/TraceLocations/TraceLocationsCoordinator.swift

```diff
@@ -115,9 +116,12 @@ class TraceLocationsCoordinator {
 				}
 			),
 			onDismiss: { [weak self] didConsent in
-				self?.infoScreenShown = didConsent
-				if !didConsent {
-					self?.parentNavigationController?.popViewController(animated: false)
+				guard let self = self else {
+					return
+				}
+				self.infoScreenShown = didConsent
+				if !alreadyDidConsentOnce && !didConsent {
+					self.parentNavigationController?.popViewController(animated: false)
 				}
 				navigationController.dismiss(animated: true, completion: nil)
 			}
```

Comment:
```suggestion
				self?.infoScreenShown = didConsent

				if !(alreadyDidConsentOnce || didConsent) {
					self?.parentNavigationController?.popViewController(animated: false)
				}
				navigationController.dismiss(animated: true, completion: nil)
			}
```
maybe this makes a bit more readable